app/assets/src/components/views/amr_heatmap/AMRHeatmapView.jsx

```diff
@@ -0,0 +1,132 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { StickyContainer, Sticky } from "react-sticky";
+
+import AMRHeatmapControls from "~/components/views/amr_heatmap/AMRHeatmapControls";
+import AMRHeatmapVis from "~/components/views/amr_heatmap/AMRHeatmapVis";
+import ErrorBoundary from "~/components/ErrorBoundary";
+import { getAMRCounts } from "~/api/amr";
+import { ViewHeader, NarrowContainer } from "~/components/layout";
+
+import cs from "./amr_heatmap_view.scss";
+
+const METRICS = [
+  { text: "Coverage", value: "coverage" },
+  { text: "Depth", value: "depth" },
+];
+
+const VIEW_LEVELS = [
+  { text: "Genes", value: "gene" },
+  { text: "Alleles", value: "allele" },
+];
+
+export default class AMRHeatmapView extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      loading: true,
+      selectedOptions: {
+        metric: "coverage",
+        viewLevel: "gene",
+      },
+    };
+  }
+
+  componentDidMount() {
+    this.requestAMRCountsData(this.props.sampleIds);
+  }
+
+  async requestAMRCountsData(sampleIds) {
+    const rawSampleData = await getAMRCounts(sampleIds);
+    const samplesWithAMRCounts = rawSampleData.filter(
+      sampleData => sampleData.error === ""
+    );
+    this.setState({
+      rawSampleData,
+      samplesWithAMRCounts,
+      sampleIds,
+      loading: false,
+    });
+  }
+
+  assembleControlOptions() {
+    // A Map is used here to preserve the order that the filters will be
+    // presented in the controls element
+    return new Map([
+      ["viewLevel", { options: VIEW_LEVELS, label: "View Level" }],
+      ["metric", { options: METRICS, label: "Metric" }],
+    ]);
+  }
+
+  updateOptions = options => {
+    let newOptions = Object.assign({}, this.state.selectedOptions, options);
+    this.setState({
+      selectedOptions: newOptions,
+    });
+  };
+
+  renderVisualization() {
+    if (this.state.loading === true) {
```

Comment:
Nit: doesn't seem that `this.state.loading` will be set as anything besides a boolean (unless I'm missing something), so can just put `if (this.state.loading) { ... }` (+ similar below)

Reply:
It is supposed to be just a boolean, I just like being explicit about it since JS is dynamically typed

Reply:
Ah never mind, AirBnB style guide says to just do `if (this.state.loading) {...}` so I'll fix it